File: Leetcode/58. Length of Last Word/solution.c
int lengthOfLastWord(char * s){
    char * p = s;
    int counter = 0;
    
    while(*p != '\0') {p++;}  // Traverse the string to the last char.
    
    // go back to count the last word.
    while(p != s) {
        if(counter && *p == ' ') break;
        if(*--p != ' ') counter++;
    }
    
    return counter;
}

File: Leetcode/58. Length of Last Word/solution.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        counter = 0
        length = 0
        for c in s:
            if c != ' ':
                counter += 1
            else:
                if counter:
                    length = counter
                counter = 0

        return length if not counter else counter
